packages/core/strapi/ee/index.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+const fs = require('fs');
+const { join } = require('path');
+const crypto = require('crypto');
+const fetch = require('node-fetch');
+const { pick } = require('lodash/fp');
+
+const { coreStoreModel } = require('../lib/services/core-store');
+const { getRecurringCronExpression } = require('../lib/utils/cron');
+
+const publicKey = fs.readFileSync(join(__dirname, 'resources/key.pub'));
+
+const ee = {
+  enabled: false,
+  licenseInfo: {},
+};
+
+const disable = (message = 'Invalid license. Starting in CE.') => {
+  ee.logger?.warn(message);
+  // Only keep the license key for potential re-enabling during a later check
+  ee.licenseInfo = pick('licenseKey', ee.licenseInfo);
+  ee.enabled = false;
+};
+
+const readLicense = (directory) => {
+  try {
+    const path = join(directory, 'license.txt');
+    return fs.readFileSync(path).toString();
+  } catch (error) {
+    if (error.code !== 'ENOENT') {
+      // Permission denied, directory found instead of file, etc.
+    }
+  }
+};
+
+const fetchLicense = async (key, fallback) => {
+  const fallbackToStoredLicense = () => {
+    const error = 'Could not proceed to the online verification of your license.';
+
+    if (fallback) {
+      ee.logger(`${error} We will try to use the locally stored one as a potential fallback.`);
+      return fallback;
+    }
+
+    disable(`${error} Sorry for the inconvenience. Starting in CE.`);
+    return null;
+  };
+
+  try {
+    const response = await fetch(`https://license.strapi.io/api/licenses/${key}`);
+
+    if (response.status >= 500) {
+      return fallbackToStoredLicense();
+    }
+
+    if (response.status >= 400) {
```

Comment:
For context, I considered using an `Api-Key` header or a custom `Authorization` header format to use the license key as an auth mechanism on the registry.

But I instead decided to keep it simple with a POST request that will return error details in case of failure as you suggested, right now the error object returned will contain the status and a message, but a specific code can be added later.